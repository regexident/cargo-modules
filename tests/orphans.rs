#[macro_use]
mod util;

mod help {
    test_cmd!(
        args: "orphans \
                --help",
        success: true,
        color_mode: ColorMode::Plain,
        project: smoke
    );
}

mod colors {
    mod plain {
        test_cmd!(
            args: "orphans",
            success: false,
            color_mode: ColorMode::Plain,
            project: orphans
        );
    }

    mod ansi {
        test_cmd!(
            args: "orphans",
            success: false,
            color_mode: ColorMode::Ansi,
            project: orphans
        );
    }

    mod truecolor {
        test_cmd!(
            args: "orphans",
            success: false,
            color_mode: ColorMode::TrueColor,
            project: orphans
        );
    }
}

mod no_orphans {
    test_cmd!(
        args: "orphans",
        success: true,
        color_mode: ColorMode::Plain,
        project: no_orphans
    );
}

mod orphans {
    test_cmd!(
        args: "orphans",
        success: false,
        color_mode: ColorMode::Plain,
        project: orphans
    );
}
