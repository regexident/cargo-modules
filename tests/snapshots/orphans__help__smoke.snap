---
source: tests/orphans.rs
expression: output
---
CMD:
orphans
--help

STDERR:

STDOUT:
Detects unlinked source files within a crate's directory.

Usage: cargo-modules orphans [OPTIONS]

Options:
      --verbose                        Use verbose output
      --lib                            Process only this package's library
      --bin <BIN>                      Process only the specified binary
  -p, --package <PACKAGE>              Package to process (see `cargo help pkgid`)
      --no-default-features            Do not activate the `default` feature
      --all-features                   Activate all available features
      --features <FEATURES>            List of features to activate. This will be ignored if `--cargo-all-features` is provided
      --target <TARGET>                Analyze for target triple
      --manifest-path <MANIFEST_PATH>  Path to Cargo.toml [default: .]
      --deny                           Returns a failure code if one or more orphans are found
      --cfg-test                       Analyze with `#[cfg(test)]` enabled (i.e as if built via `cargo test`)
  -h, --help                           Print help
